xonsh/history.py

```diff
@@ -333,77 +511,88 @@ def flush(self, at_exit=False):
         self.buffer.clear()
         return hf
 
+    @staticmethod
+    def show(ns=None, hist=None, start_index=None, end_index=None,
```

Comment:
This seems like it should be a function in the global namespace, not a staticmethod.  It seems deeply tied to the CLI and not the history class itself.